mob/src/main/java/com/github/phantazmnetwork/mob/spawner/BasicMobSpawner.java

```diff
@@ -3,64 +3,169 @@
+    public BasicMobSpawner(@NotNull Map<BooleanObjectPair<String>, ConfigProcessor<?>> processorMap,
+            @NotNull MobStore mobStore, @NotNull Spawner neuralSpawner, @NotNull ContextManager contextManager,
+            @NotNull KeyParser keyParser) {
+        this.processorMap = Map.copyOf(processorMap);
         this.mobStore = Objects.requireNonNull(mobStore, "mobStore");
         this.neuralSpawner = Objects.requireNonNull(neuralSpawner, "neuralSpawner");
+        this.contextManager = Objects.requireNonNull(contextManager, "contextManager");
+        this.keyParser = Objects.requireNonNull(keyParser, "keyParser");
     }
 
     @Override
     public @NotNull PhantazmMob spawn(@NotNull Instance instance, @NotNull Point point, @NotNull MobModel model) {
         NeuralEntity neuralEntity = neuralSpawner.spawnEntity(instance, point, model.getDescriptor());
-        setDisplayName(neuralEntity, model);
+        setEntityMeta(neuralEntity, model);
         setEquipment(neuralEntity, model);
         setAttributes(neuralEntity, model);
         setHealth(neuralEntity, model);
 
-        PhantazmMob mob = new PhantazmMob(model, neuralEntity);
-        addGoals(mob);
+        Module module = new Module(neuralEntity);
+        DependencyProvider dependencyProvider = new ModuleDependencyProvider(module, keyParser);
+        ElementContext context = contextManager.makeContext(model.getNode());
+        addGoals(context, dependencyProvider, neuralEntity, model);
+        Map<Key, Collection<Skill>> triggers = createTriggers(context, dependencyProvider, neuralEntity, model);
 
+        PhantazmMob mob = new PhantazmMob(model, neuralEntity, triggers);
         mobStore.registerMob(mob);
         return mob;
     }
 
-    private void setDisplayName(@NotNull NeuralEntity neuralEntity, @NotNull MobModel model) {
-        model.getDisplayName().ifPresent(displayName -> {
-            neuralEntity.setCustomName(displayName);
-            neuralEntity.setCustomNameVisible(true);
-        });
+    private void setEntityMeta(@NotNull NeuralEntity neuralEntity, @NotNull MobModel model) {
+        EntityMeta meta = neuralEntity.getEntityMeta();
+        ConfigNode metaNode = model.getMetaNode();
+        for (Method method : meta.getClass().getMethods()) {
+            if (!Modifier.isPublic(method.getModifiers()) || method.getReturnType() != void.class) {
+                continue;
+            }
+
+            Parameter[] parameters = method.getParameters();
+            if (parameters.length != 1) {
+                continue;
+            }
+
+            String methodName = method.getName();
+            if (!methodName.startsWith("set")) {
+                continue;
+            }
+            String key;
+            if (methodName.length() > 3) {
+                key = Character.toLowerCase(methodName.charAt(3)) + methodName.substring(4);
+            }
+            else {
+                key = methodName.substring(3);
```

Comment:
since this statement appears to only be hit when methodName is length 3, key always becomes an empty string here
it can't be less than 3 because it must start with "set"
and if it's greater than 3 the first if is hit
don't know if you should consider "" to be a valid key so might be good to continue here?